src/corefx/System.Globalization.Native/icushim.cpp

```diff
@@ -170,6 +170,13 @@ bool FindLibWithMajorMinorSubVersion(int* majorVer, int* minorVer, int* subVer)
     return false;
 }
 
+static int32_t g_icuPresent = 1;
```

Comment:
It may not actually matter, but did you consider reversing this? i.e. starting it at 0 == not present, and then setting it to 1 only once ICU has been detected and all of the function pointers appropriately initialized?